pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -347,19 +349,21 @@ public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingCon
         }
         indexDir = downloadSegment(segmentName, zkMetadata);
       } else {
-        LOGGER.info("Reload existing segment: {} of table: {}", segmentName, _tableNameWithType);
+        LOGGER.info("Reload existing segment: {} of table: {} currently on tier: {}", segmentName, _tableNameWithType,
+            segmentTier);
         // The indexDir is empty after calling createBackup, as it's renamed to a backup directory.
         // The SegmentDirectory should initialize accordingly. Like for SegmentLocalFSDirectory, it
         // doesn't load anything from an empty indexDir, but gets the info to complete the copyTo.
         try (SegmentDirectory segmentDirectory = initSegmentDirectory(segmentName, String.valueOf(zkMetadata.getCrc()),
-            indexLoadingConfig, schema)) {
+            segmentTier, indexLoadingConfig, schema)) {
           segmentDirectory.copyTo(indexDir);
         }
       }
 
       // Load from indexDir and replace the old segment in memory. What's inside indexDir
       // may come from SegmentDirectory.copyTo() or the segment downloaded from deep store.
-      ImmutableSegment segment = ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, schema);
+      ImmutableSegment segment =
```

Comment:
Consider wrapping `tableDataDir` and `tier` into the `indexLoadingConfig` to reduce the parameters passed.